bot/exts/info/doc/_parsing.py

```diff
@@ -0,0 +1,328 @@
+    """
+    Truncate passed signatures to not exceed `_MAX_SIGNAUTRES_LENGTH`.
+
+    If the signatures need to be truncated, parameters are collapsed until they fit withing the limit.
+    Individual signatures can consist of max 1, 2, ..., `_MAX_SIGNATURE_AMOUNT` lines of text,
+    inversely proportional to the amount of signatures.
+    A maximum of `_MAX_SIGNATURE_AMOUNT` signatures is assumed to be passed.
+    """
+    if not sum(len(signature) for signature in signatures) > _MAX_SIGNATURES_LENGTH:
+        return signatures
+
+    max_signature_length = _EMBED_CODE_BLOCK_LINE_LENGTH * (_MAX_SIGNATURE_AMOUNT + 1 - len(signatures))
+    formatted_signatures = []
+    for signature in signatures:
+        signature = signature.strip()
+        if len(signature) > max_signature_length:
+            if (parameters_match := _PARAMETERS_RE.search(signature)) is None:
+                formatted_signatures.append(textwrap.shorten(signature, max_signature_length))
+                continue
+
+            truncated_signature = []
+            parameters_string = parameters_match[1]
+            running_length = len(signature) - len(parameters_string)
+            for parameter in _split_parameters(parameters_string):
+                if (len(parameter) + running_length) <= max_signature_length - 4:  # account for comma and placeholder
+                    truncated_signature.append(parameter)
+                    running_length += len(parameter) + 1
+                else:
+                    truncated_signature.append(" ...")
+                    formatted_signatures.append(signature.replace(parameters_string, ",".join(truncated_signature)))
+                    break
+        else:
+            formatted_signatures.append(signature)
+
+    return formatted_signatures
+
+
+def _get_truncated_description(
+        elements: Iterable[Union[Tag, NavigableString]],
+        markdown_converter: DocMarkdownConverter,
+        max_length: int,
+        max_lines: int,
+) -> str:
+    """
+    Truncate markdown from `elements` to be at most `max_length` characters when rendered or `max_lines` newlines.
+
+    `max_length` limits the length of the rendered characters in the string,
+    with the real string length limited to `_MAX_DESCRIPTION_LENGTH` to accommodate discord length limits
+    """
+    result = ""
+    markdown_element_ends = []
+    rendered_length = 0
+
+    tag_end_index = 0
+    for element in elements:
+        is_tag = isinstance(element, Tag)
+        element_length = len(element.text) if is_tag else len(element)
+
+        if rendered_length + element_length < max_length:
+            if is_tag:
+                element_markdown = markdown_converter.process_tag(element)
+            else:
+                element_markdown = markdown_converter.process_text(element)
+
+            element_markdown_length = len(element_markdown)
+            rendered_length += element_length
+            tag_end_index += element_markdown_length
```

Comment:
Should we attempt to go deeper in the handling? Currently in case we get a long tag like a div or ul with different p tags we'll most probably just skip it while we would be able to extract the p tags and split them properly but because we'd be keeping the end of the tags out it'd need some per tag behaviour complicating the implementation even more (open to suggestions here as I feel like it is a bit of a mess)

Reply:
Maybe do this recursively for certain types of nested tags?

Reply:
That does sound like a good idea, but I'm not sure about the implementation. Calling this again is not viable as we then don't have access to the rendered length and it'll probably need separate handling for each type of tag that we'll want to handle recursively. Maybe making a class for this to share the state and the methods would be the best.

Reply:
After thinking about it more I've decided it's probably not worth the added complexity, the symbols pages of symbols I used when testing seem to be mostly fine and the whole text is always a few clicks away.